src/checkers/Cashflow.ml

```diff
@@ -1604,21 +1630,38 @@ module ScillaCashflowChecker
                   cf_tag_expr e (lub_tags (get_id_tag f) NoInfo) (AssocDictionary.make_dict ()) (AssocDictionary.make_dict ()) ctr_tag_map in
              AssocDictionary.insert (get_id f) e_tag acc_env)
       in
-      let rec tagger cconstraint components param_env field_env ctr_tag_map =
+    let rec tagger cconstraint (fields:(field list)) components param_env field_env ctr_tag_map =
         let (new_constraint, tmp_param_env, _, tmp_ctr_tag_map, constraint_changes) =
           cf_tag_expr cconstraint NotMoney param_env (AssocDictionary.make_dict()) ctr_tag_map in
-        let (new_ts, new_param_env, new_field_env, tmp_ctr_tag_map, changes) =
-          List.fold_right components ~init:([], tmp_param_env, field_env, tmp_ctr_tag_map, constraint_changes) 
+
+      (* adding field tagging *)
+      let (new_fs, new_param_env1, new_field_env1, tmp_ctr_tag_map, field_changes) =
+        (* fold right for backwards analysis *)
+        List.fold_right fields ~init:([], tmp_param_env, field_env, tmp_ctr_tag_map, constraint_changes) 
+          ~f:(fun f (acc_fs, acc_param_env, acc_field_env, acc_ctr_tag_map, acc_changes) ->
+              let (new_f, new_param_env, new_field_env, new_ctr_tag_map, changes) =
+                cf_tag_field f acc_param_env acc_field_env acc_ctr_tag_map in
+              (new_f :: acc_fs, new_param_env, new_field_env, new_ctr_tag_map, acc_changes || changes))
+      in
+
+
+      (* for all components traverse their stmts until no change *)
+      let (new_ts, new_param_env2, new_field_env2, tmp_ctr_tag_map, acc_components_changes) =
```

Comment:
Same issue here: `tmp_ctr_tag_map` should be `new_ctr_tag_map2`, and `acc_components_changes` should just be `changes`.